### DISCUSSION TITLE
Motionplus: stream output interfacing with apps/Kodi

### DISCUSSION THREAD

**Discussion by nightcustard** [Motion-Project/motionplus#200] [Q&A]
I'm a long time Raspberry Pi Motion user and use Motion for security cameras and a nest box.  I display the Motion mjpeg stream output on an Android app called tinyCam Pro, which has a specific option for Motion streamed video (server push mjpeg).  I would also like to use Kodi to display the mjpeg stream output but this is currently unusable as the video keeps buffering due to the low frame rate.  I have therefore recently tried a Motionplus install to see if an mpgts stream output would work any better with Kodi.  As it turns out, the current stream output is incompatible with both tinyCam Pro and Kodi stream inputs due to the html wrapping.

Is it possible the Motionplus video stream could be made available without the html wrapping, and even better the mjpeg and mpgts outputs be made available on different ports?

Incidentally, I've noticed the web interface appears to currently ignores the camera element of the web address (ie) http://192.168.1.45:8081/1/mjpeg gives me the same low resolution display as does http://192.168.1.45:8081/0/mjpeg.  I've tried this on both Chrome and Firefox browsers with no differences.

I've installed MotionPlus 0.2.0-git-20241121-3e28491 on a Pi 4 running bookworm lite and a Pi camera v3.  The install script works very well, BTW!

**Comment by Mr-Dave** [comment]
Discussion moved from Motion repo.

> As it turns out, the current stream output is incompatible with both tinyCam Pro and Kodi stream inputs due to the html wrapping.

There is not any html wrapping.  I do not know about TinyCam Pro but for Kodi and other applications such as mpv they can play the mpegts feed...But...the apps just don't like the 1fps at all.  The `stream_maxrate` needs to be something higher (I realize this is probably beyond the ability of  PIs)

**Reply by nightcustard** [reply]
Thanks for the speedy response! I understand your comments about the low FPS available from Raspberry Pis and its incompatibility with Kodi - 1fps is the best I've achieved with my Pi 3s. 
As the the wrapper comment, I guess I must apologise for using terms I only vaguely understand(!). With Motion, when I enter the IP address:8081 into a browser, I get a video display. When I enter the MotionPlus equivalent (IP address:8081/1/mjpeg), I get a web page with a video window inside it. I then have to select my camera from the drop down menu to get the full res video. This is what I meant by an html wrapper.  Is there a way in MotionPlus to get the same stream output I get from Motion?

**Reply by Mr-Dave** [reply]
The base streams are available in Motionplus and described in the [examples](https://motion-project.github.io/motionplus_examples.html#webcontrol_pages)  (When I specify `webcontrol_port 8081` and `camera_id 1`, the address `http://localhost:8080/1/mjpg/` does provide a feed without a webpage.  Note the spelling difference of mjpg and not mjpeg)

**Reply by nightcustard** [reply]
Ah! Excellent - this is what I was hoping for and I guess I should have known it would need a config entry.   I'll give it a go shortly. Thanks again.
